R/ggMarginal.R

```diff
@@ -174,73 +120,76 @@ ggMarginal <- function(p, data, x, y, type = c("density", "histogram", "boxplot"
   # - Remove all lines and colours from marginal plots
   # - Use the same axis titles as the main plot, to ensure the same space is taken
   # - Use the same axis range as the main plot
-  if (margins != "y") {
-    top <- marginPlot(margin = "x", type = type, xvar = xvar, yvar = yvar, 
-                      xparams = xparams, yparams = yparams, pb = pb, data = data,
-                      extraParams = extraParams)
-    top <- addMainTheme(marginal = top, margin = "x", p = p)
-    top <- top +
-      ggplot2::ylab(p$labels$y) +
-      getScale(margin = "x", type = type, pb = pb)
+  
+  # ... If margins = x or margins = (x and y), then you have to create top plot
+  # Top plot = horizontal margin plot, which corresponds to x marg
+  if (margins != "y") { 
+    rawTop <- genMargePlot(marg = "x", type = type, scatPbuilt = scatPbuilt, 
```

Comment:
This will sound petty, but can you change `genMargePlot` to `genMargPlot`? "Marge" is an incorrect spelling, it's not shorthand for anything, either "marg" or "marginal" :)